### PR TITLE
Export to excel

### PR DESCRIPTION
**Pull request by vinay-easwaran**

### CODE REVIEW

File: htdocs/Language/default.php [hunk 1]
```diff
@@ -217,7 +217,9 @@
 		"HEIGHT" => "Height",
 		"FONT_SIZE" => "Font Size",
 		"COL_COUNT" => "Column-Count",
-		"CMD_REFRESH" => "Refresh"
+		"CMD_REFRESH" => "Refresh",
+		"SPECIMEN_CUSTOM_FIELDS" => "Specimen Custom Fields",
+		"PATIENT_CUSTOM_FIELDS" => "Patient Custom Fields"
```
Comment (mrysav) [inline htdocs/Language/default.php:222]: You did this correctly in the PHP files, but please also update `default.xml` and `en.xml` as well.

File: htdocs/export/export_to_excel.php [hunk 2]
```diff
@@ -122,10 +131,22 @@
...
 EOQ;
 
+    $results = query_associative_all($query);
 
-    $sheet = $objPHPExcel->createSheet();
+    if(count($results) == 0) {
+        header('HTTP/1.1 404 Not Found', true, 404);
+        echo "No data to report in the chosen date range";
+        exit;
+    }
```
Comment (mrysav) [inline htdocs/export/export_to_excel.php:140]: Ah - good change!

File: htdocs/reports/reports_dailypatients.php [hunk 3]
```diff
@@ -53,13 +53,31 @@
 	$margin_list[$i] = ($SCREEN_WIDTH * $margin_list[$i] / 100);
 }
 ?>
+<script src="js/jquery.min.js"></script>
+<script src="js/jquery.table2excel.js"></script>
```
Comment (mrysav) [inline htdocs/reports/reports_dailypatients.php:57]: thank you!

File: htdocs/reports/reports_dailypatients.php [hunk 4]
```diff
@@ -53,13 +53,31 @@
...
+{
+	var content = $('#'+div_id).html();
+
+	const date = new Date();
+	const formattedDate = `${date.getFullYear()}${("0" + (date.getMonth()+1)).slice(-2)}${("0" + (date.getDate())).slice(-2)}${date.getHours()}${date.getMinutes()}`;
+
+	var fileName = `blisreport_${formattedDate}.xls`;
+
+	$("#report_content_table5").table2excel({
+		// exclude: ".excludeThisClass",
```
Comment (mrysav) [inline htdocs/reports/reports_dailypatients.php:75]:
```suggestion

```

File: htdocs/reports/reports_dailypatients.php [hunk 5]
```diff
@@ -53,13 +53,31 @@
...
+		name: "LangUtil::$generalTerms['PATIENT_DAILY_LOG_SHEET_NAME']",
```
Comment (mrysav) [inline htdocs/reports/reports_dailypatients.php:76]: I think this is in the Javascript code, so you have to do something like this, right?

```suggestion
		name: "<?php echo(LangUtil::$generalTerms['PATIENT_DAILY_LOG_SHEET_NAME']) ?>",

```

File: htdocs/reports/reports_dailyspecimens.php [hunk 6]
```diff
@@ -120,13 +120,31 @@ function get_records_to_print($lab_config, $test_type_id, $date_from, $date_to)
...
+	var content = $('#'+div_id).html();
+
+	const date = new Date();
+	const formattedDate = `${date.getFullYear()}${("0" + (date.getMonth()+1)).slice(-2)}${("0" + (date.getDate())).slice(-2)}${date.getHours()}${date.getMinutes()}`;
+
+	var fileName = `blisreport_${formattedDate}.xls`;
+
+	$("#report_content_table4").table2excel({
+		// exclude: ".excludeThisClass",
+		name: "LangUtil::$generalTerms['SPECIMEN_DAILY_LOG_SHEET_NAME']",
```
Comment (mrysav) [inline htdocs/reports/reports_dailyspecimens.php:143]:
```suggestion
		name: "<?php echo(LangUtil::$generalTerms['SPECIMEN_DAILY_LOG_SHEET_NAME']) ?>",

```

File: local/langdata_127/default.php [hunk 7]
```diff
@@ -617,7 +619,9 @@
 		"INCLUDE_PATIENT_NAME" => "Include patient name",
 		"INCLUDE_PATIENT_BIRTHDATE" => "Include patient birth date",
 		"INCLUDE_PATIENT_SEX" => "Include patient sex",
-        "INCLUDE_PATIENT_ID" => "Include patient ID"
```
Comment (mrysav) [inline local/langdata_127/default.php:620]: Can you also update _all_ the XML files where you changed the PHP files? (You can copy paste)

Recent experience on my end shows that we should keep local/langdata_127, langdata_12, and langdata_revamp in sync.

Comment (vinay-easwaran) [reply local/langdata_127/default.php:620]: Updated

**Review by mrysav — APPROVED** [review]
I'm going to go ahead and merge this! Thanks everyone!

**Pull request merged**